src/openforms/authentication/contrib/generic_oidc/backends.py

```diff
@@ -0,0 +1,194 @@
+from mozilla_django_oidc_db.config import dynamic_setting
+from mozilla_django_oidc_db.utils import obfuscate_claims
+from mozilla_django_oidc_db.views import _RETURN_URL_SESSION_KEY
+
+from openforms.forms.models import Form, FormAuthenticationBackend
+from openforms.typing import JSONObject
+
+from ..yivi_oidc.constants import PLUGIN_ID as YIVI_PLUGIN_ID
+from ..yivi_oidc.models import AttributeGroup, YiviOpenIDConnectConfig
+from .plugin import get_config_to_plugin
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class GenericOIDCBackend(OIDCAuthenticationBackend):
+    """
+    A generic backend specialised to the OF OIDC plugins.
+    """
+
+    OF_OIDCDB_REQUIRED_CLAIMS = dynamic_setting[list[str]](default=[])
+
+    @override
+    def _check_candidate_backend(self) -> bool:
+        # if we're dealing with a mozilla-django-oidc-db config that is *not* a
+        # OF config, then don't bother.
+        if issubclass(self.config_class, BaseConfig) or issubclass(
+            self.config_class, YiviOpenIDConnectConfig
+        ):
+            return super()._check_candidate_backend()
+        return False
+
+    def get_or_create_user(
+        self, access_token: str, id_token: str, payload: JSONObject
+    ) -> AnonymousUser:
+        """
+        Return a "fake" Django user.
+
+        If the claims are valid, we only process them and do not create or update an
+        actual Django user.
+        """
+        assert isinstance(self.request, HttpRequest)
+
+        user_info = self.get_userinfo(access_token, id_token, payload)
+        if not self.verify_claims(user_info):
+            # Raise PermissionDenied rather than SuspiciousOperation - this makes it
+            # Django stops trying other (OIDC) authentication backends, which fail
+            # because the code was already exchanged for an access token.
+            # Note that this backend only runs for the OF OIDC configs at all,
+            # and those aren't particularly compatible with the admin-OIDC flow anyway.
+            # See :meth:`_check_candidate_backend` that prevents this backend from being
+            # used for admin OIDC.
+            raise PermissionDenied("Claims verification failed")
+
+        self._extract_and_store_claims(payload)
+
+        user = AnonymousUser()
+        user.is_active = True  # type: ignore
+        return user
+
+    def _process_claims(self, claims: JSONObject) -> JSONObject:
+        # see if we can use a cached config instance from the settings configuration
+        assert hasattr(self, "_config") and (
+            isinstance(self._config, BaseConfig)
+            or isinstance(self._config, YiviOpenIDConnectConfig)
+        )
+
+        # Because Yivi has loa config for BSN and KVK, we need to set the "global" loa
+        # config dynamically based on which data is presented.
+        # This will be moved after oidc-db changes.
+        if issubclass(self.config_class, YiviOpenIDConnectConfig):
```

Comment:
Is there a way we can push this code to the plugin? This is a big OOP violation - suddenly your base class/module/package needs to know about more concrete subclasses/derivates.